File: P31.Abdul Bari/L01.arrays/Rectangle.java
public class Rectangle {
    int length;
    int breadth;
    char x;

    public static void main(String[] args) {
        Rectangle r1 = new Rectangle();
        r1.length = 10;
        r1.breadth = 5;

        r1.length = 15;
        r1.breadth = 7;

        System.out.println(r1.length);
        System.out.println(r1.breadth);
    }
}

/*
#include <iostream>

using namespace std;

struct Rectangle {
    int length;
    int breadth;
    char x;
};

int main() {
    struct Rectangle r1 = {10, 5};

    r1.length = 15;
    r1.breadth = 7;

    cout << r1.length << endl;
    cout << r1.breadth << endl;

    return 0;
}
    

#include <stdio.h>

struct Rectangle {
    int length;
    int breadth;
    char x;
};

int main() {
    struct Rectangle r1 = {10, 5};

    // Modify the values
    r1.length = 15;
    r1.breadth = 7;

    // Print the values
    printf("Length: %d\n", r1.length);
    printf("Breadth: %d\n", r1.breadth);

    return 0;
}
    
class Rectangle:
    def __init__(self, length, breadth, x):
        self.length = length
        self.breadth = breadth
        self.x = x

    def print_dimensions(self):
        print("Length:", self.length)
        print("Breadth:", self.breadth)

r1 = Rectangle(10, 5, 'A')

# Modify the values
r1.length = 15
r1.breadth = 7

# Print the values
r1.print_dimensions()
*/

File: P31.Abdul Bari/L00.BasicsCPP/Rectangle.cpp
// #include <iostream>
// using namespace std;

// class Rectangle {
// private:
//     int length;
//     int breadth;

// public:
//     // Default constructor
//     Rectangle() {
//         length = 1;
//         breadth = 1;
//     }

//     // Parameterized constructor
//     Rectangle(int l, int b) {
//         length = l;
//         breadth = b;
//     }

//     // Member function to calculate area
//     int area() {
//         return length * breadth;
//     }

//     // Member function to calculate perimeter
//     int perimeter() {
//         return 2 * (length + breadth);
//     }

//     // Getter for length
//     int getLength() {
//         return length;
//     }

//     // Setter for length
//     void setLength(int l) {
//         length = l;
//     }

//     // Destructor
//     ~Rectangle() {
//         // Optional destructor body (automatically called)
//     }
// };

// int main() {
//     Rectangle r(10, 5); // Create an object of Rectangle with initial values
//     cout << r.area() << endl;           // Print area
//     cout << r.perimeter() << endl;      // Print perimeter
//     r.setLength(20);                    // Change the length
//     cout << r.getLength() << endl;      // Print the updated length

//     return 0;
// }



#include <iostream>
using namespace std;

class Rectangle {
private:
    int length;
    int breadth;

public:
    // Default constructor
    Rectangle();

    // Parameterized constructor
    Rectangle(int l, int b);

    // Member function to calculate area
    int area();

    // Member function to calculate perimeter
    int perimeter();

    // Getter for length
    int getLength();

    // Setter for length
    void setLength(int l);

    // Destructor
    ~Rectangle();
};

// Default constructor implementation
Rectangle::Rectangle() {
    length = 1;
    breadth = 1;
}

// Parameterized constructor implementation
Rectangle::Rectangle(int l, int b) {
    length = l;
    breadth = b;
}

// Function to calculate area
int Rectangle::area() {
    return length * breadth;
}

// Function to calculate perimeter
int Rectangle::perimeter() {
    return 2 * (length + breadth);
}

// Getter for length
int Rectangle::getLength() {
    return length;
}

// Setter for length
void Rectangle::setLength(int l) {
    length = l;
}

// Destructor
Rectangle::~Rectangle() {
    // Optional cleanup if needed
}

int main() {
    Rectangle r(10, 5); // Create an object of Rectangle with initial values
    cout << r.area() << endl;           // Print area
    cout << r.perimeter() << endl;      // Print perimeter
    r.setLength(20);                    // Change the length
    cout << r.getLength() << endl;      // Print the updated length

    return 0;
}
